docs/src/main/asciidoc/application-data-caching-guide.adoc

```diff
@@ -0,0 +1,474 @@
+
+[source,java]
+----
+package org.acme.caching;
+
+import java.util.List;
+
+public class WeatherForecast {
+
+    private List<String> dailyForecasts;
+
+    private long executionTimeInMs;
+
+    public WeatherForecast(List<String> dailyForecasts, long executionTimeInMs) {
+        this.dailyForecasts = dailyForecasts;
+        this.executionTimeInMs = executionTimeInMs;
+    }
+
+    public List<String> getDailyForecasts() {
+        return dailyForecasts;
+    }
+
+    public long getExecutionTimeInMs() {
+        return executionTimeInMs;
+    }
+
+    // Setters omitted.
+}
+----
+
+Now, we just need to update the generated `WeatherForecastResource` class to use the service and response:
+
+[source,java]
+----
+package org.acme.caching;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+@Path("/weather")
+public class WeatherForecastResource {
+
+    @Inject
+    WeatherForecastService service;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public WeatherForecast getForecast(@QueryParam("city") String city, @QueryParam("daysInFuture") long daysInFuture) { <1>
+        long executionStart = System.currentTimeMillis();
+        List<String> dailyForecasts = Arrays.asList(
+                service.getDailyForecast(LocalDate.now().plusDays(daysInFuture), city),
+                service.getDailyForecast(LocalDate.now().plusDays(daysInFuture + 1L), city),
+                service.getDailyForecast(LocalDate.now().plusDays(daysInFuture + 2L), city)
+        );
+        long executionEnd = System.currentTimeMillis();
+        return new WeatherForecast(dailyForecasts, executionEnd - executionStart);
+    }
+}
+----
+<1> If the `daysInFuture` query parameter is omitted, the three days weather forecast will start from the current day.
+Otherwise, it will start from the current day plus the `daysInFuture` value.
+
+We're all done! Let's check if everything's working.
+
+First, run the application using `./mvnw compile quarkus:dev` from the project directory.
+
+Then, call `http://localhost:8080/weather?city=Raleigh` from a browser. After six long seconds, the application will answer
+something like this:
+
+[source]
+----
+{"dailyForecasts":["MONDAY will be cloudy in Raleigh","TUESDAY will be chilly in Raleigh","WEDNESDAY will be rainy in Raleigh"],"executionTimeInMs":6001}
+----
+
+[TIP]
+====
+The response content may vary depending on the day you run the code.
+====
+
+You can try calling the same URL again and again, it will always take six seconds to answer.
+
+== Enabling the cache
+
+Now that your Quarkus application is up and running, let's tremendously improve its response time by caching the external
+meteorological service responses. Update the `WeatherForecastService` class like this:
```

Comment:
No need to change it but for the future: we try to have max one sentence per line in our doc. You can split a sentence on several lines if needed but we should avoid having 2 sentences per line.
Asciidoc will join everything anyway.

It's easier to parse when reading and it makes future diffs much more readable.

Reply:
Ok I didn't know that. Should a doc like be limited to 128 characters (like the limit enforced for the code)?

Reply:
Not really. We don't have a strong convention for the doc. The idea is just to keep them readable.

The max one sentence per line is mostly there for having nicer diffs in the future. It's not mandatory but nice to have. You'll see, it helps a lot when reviewing doc changes.

Reply:
Split done :)